src/components/Details/DetailedTrafficList.tsx

```diff
@@ -0,0 +1,261 @@
+  traffic: TrafficItem[];
+};
+
+export interface TrafficDataHttp {
+  protocol: 'http';
+  rates: {
+    http: string;
+    httpPercentErr?: string;
+  };
+}
+
+export interface TrafficDataGrpc {
+  protocol: 'grpc';
+  rates: {
+    grpc: string;
+    grpcPercentErr?: string;
+  };
+}
+
+export interface TrafficDataTcp {
+  protocol: 'tcp';
+  rates: {
+    tcp: string;
+  };
+}
+
+export type TrafficData = TrafficDataHttp | TrafficDataTcp | TrafficDataGrpc;
+
+export interface AppNode {
+  id: string;
+  type: NodeType.APP;
+  namespace: string;
+  name: string;
+  version: string;
+}
+
+export interface WorkloadNode {
+  id: string;
+  type: NodeType.WORKLOAD;
+  namespace: string;
+  name: string;
+}
+
+export interface ServiceNode {
+  id: string;
+  type: NodeType.SERVICE;
+  namespace: string;
+  name: string;
+}
+
+export interface UnknownNode {
+  id: string;
+  type: NodeType.UNKNOWN;
+  namespace: string;
+  name: 'unknown';
+}
+
+export type TrafficNode = WorkloadNode | ServiceNode | UnknownNode | AppNode;
+
+export interface TrafficItem {
+  node: TrafficNode;
+  proxy?: TrafficItem;
+  traffic: TrafficData;
+}
+
+const healthColumnSizes = {
+  md: 1,
+  sm: 1,
+  xs: 1
+};
+const minichartColumnSizes = {
+  md: 3,
+  sm: 3,
+  xs: 3
+};
+const workloadColumnSizes = {
+  md: 3,
+  sm: 3,
+  xs: 3
+};
+const trafficColumnSizes = workloadColumnSizes;
+const typeColumnSizes = healthColumnSizes;
+
+class DetailedTrafficList extends React.Component<DetailedTrafficProps> {
+  render() {
+    return (
+      <TableGrid id="table-grid" bordered={true} selectType="none">
+        <TableGrid.Head>
+          <TableGrid.ColumnHeader {...healthColumnSizes}>Health</TableGrid.ColumnHeader>
+          <TableGrid.ColumnHeader {...workloadColumnSizes}>
+            {this.props.direction === 'inbound' ? 'Source' : 'Destination'}
+          </TableGrid.ColumnHeader>
+          <TableGrid.ColumnHeader {...typeColumnSizes}>Type</TableGrid.ColumnHeader>
+          <TableGrid.ColumnHeader {...trafficColumnSizes}>Traffic</TableGrid.ColumnHeader>
+        </TableGrid.Head>
+        <TableGrid.Body>
+          {this.getSortedTraffic().map(item => {
+            return (
+              <TableGrid.Row key={item.node.id}>
+                {this.renderHealthColumn(item.traffic)}
+                {this.renderWorkloadColumn(item.node)}
+                {this.renderTypeColumn(item.traffic)}
+                {this.renderTrafficColumn(item.traffic)}
+                {this.renderMinichartColumn(item.traffic)}
+              </TableGrid.Row>
+            );
+          })}
+        </TableGrid.Body>
+      </TableGrid>
+    );
+  }
+
+  private renderHealthColumn = (traffic: TrafficData) => {
+    if (traffic.protocol === 'tcp') {
+      return (
+        <TableGrid.Col {...healthColumnSizes}>
+          <Icon type="pf" name="unknown" />
+        </TableGrid.Col>
+      );
+    } else {
+      let percentError: number;
+      if (traffic.protocol === 'http') {
```

Comment:
When loading the page just after the pods are started, I got an error below because `traffic.rates` is null. Also `traffic.protocol` was an empty string. The error was gone after some time (I guess after some metrics were filled in prometheus)

Reply:
Also I have a similar error when I'm activating traffic mirroring. When I fix this issue, another issue occurs: error rates from mirrored traffic isn't displayed. This is a known issue in the graph but IMO also raises the problem of pulling data from graph endpoint, where metrics reporting is only taken from one side, whereas a different algorithm is used in health endpoints to get a combination of both reporters and thus won't miss any data.